apps/sms/test/unit/thread_ui_test.js

```diff
@@ -1267,22 +1271,28 @@ suite('thread_ui.js >', function() {
           );
         });
 
-        test('input value is invalid ', function() {
+        test('input value is invalid ', function(done) {
           // An actual accepted recipient from contacts
           fixture.number = 'foo';
           fixture.isQuestionable = true;
 
           ThreadUI.recipients.add(fixture);
-          assert.isFalse(fixture.isInvalid);
+          /* The previous add triggers an asynchronous behavior when validating
+           * this new recipient. Let's wait for this. */
```

Comment:
I think we need to find a better way to test this; ideally I'd like that MockRecipients is dumber and especially does not emit events. But this is really painful to change now and I think it's not the goal of this patch... so here is a workaround.